src/github.com/couchbase/sync_gateway/rest/server_context.go

```diff
@@ -511,3 +513,59 @@ func (sc *ServerContext) Stats() map[string]interface{} {
 		"databases":    stats,
 	}
 }
+
+///////// ACCESS WARNINGS
+
+// If no users defined (config or bucket), issue a warning + give tips to fix
+// If guest user defined, but has no access to channels .. issue warning + tips to fix
+func emitAccessRelatedWarnings(config *DbConfig, context *db.DatabaseContext) {
+	for _, warning := range collectAccessRelatedWarnings(config, context) {
+		base.Warn("%v", warning)
+	}
+
+}
+
+func collectAccessRelatedWarnings(config *DbConfig, context *db.DatabaseContext) []string {
+
+	currentDb, err := db.GetDatabase(context, nil)
+	if err != nil {
+		base.Warn("Could not get database, skipping access related warnings")
+	}
+
+	// Query the "view principals" view which will return users and roles.  We only want to
+	// find out if there is at least one user (or role) defined, so set limit == 1 to minimize
+	// performance hit of query.
+	viewOptions := db.Body{
+		"stale": false,
+		"limit": 1,
+	}
+	vres, err := currentDb.Bucket.View(db.DesignDocSyncGateway, db.ViewPrincipals, viewOptions)
+	if err != nil {
+		base.Warn("Error trying to query ViewPrincipals: %v", err)
+		return []string{}
+	}
+
+	// if no users defined in config, and no users were returned from the view, add warning.
+	// NOTE: currently ignoring the fact that the config could contain only disabled=true users.
+	if len(config.Users) == 0 && len(vres.Rows) == 0 {
```

Comment:
I think we should check config.Users first, and so avoid the view call if len(config.Users) > 0.  That would also provide a workaround if customers with tons of users see problems with the view performance - they could just stick a disabled guest user into their config.